jscomp/bsb/bsb_ninja_gen.ml

```diff
@@ -123,6 +123,7 @@ let output_ninja_and_namespace_map
   let cwd_lib_bs = per_proj_dir // lib_artifacts_dir in 
   let ppx_flags = Bsb_build_util.ppx_flags ppx_files in
   let oc = open_out_bin (cwd_lib_bs // Literals.build_ninja) in          
+  let bsc_flags = "-bs-D BSB_BACKEND=\"js\"" :: bsc_flags in
   let g_pkg_flg , g_ns_flg = 
```

Comment:
what's the use case here?

Reply:
This is useful for code that needs different implementations on native and JS. For instance in Reprocessing we want hot reloading to work in bytecode, but not for native and JS: https://github.com/Schmavery/reprocessing/blob/eb128c16a9122a4591df3f0d81f739b2b0304dbd/src/Reprocessing_Hotreload.ml

Reply:
we already have `-bs-D BS` for JS, is it not good for use?

Reply:
I think BSB_BACKEND is a bit easier to understand than `#if BS`, maybe if it could be renamed to `JS` then it’d be cohesive with `BYTECODE` and `NATIVE`. 
Is there a downside to have one variable with multiple values, compared to multiple variables?

Reply:
Actually `JS`, `BYTECODE`, `NATIVE` are better you're right! You only need to remember that one thing instead of the key `BSB_BACKEND` and the value `"js"`. I updated the code to add `JS`.